src/backend/distributed/commands/utility_hook.c

```diff
@@ -419,8 +420,32 @@ multi_ProcessUtility(PlannedStmt *pstmt,
 	}
 
 	pstmt->utilityStmt = parsetree;
-	standard_ProcessUtility(pstmt, queryString, context,
-							params, queryEnv, dest, completionTag);
+
+	PG_TRY();
+	{
+		if (IsA(parsetree, AlterTableStmt))
```

Comment:
Is it ever possible that `activeAlterTables > 1`?

If not, in order to get rid of pg_try/pg_catch, can we simply use a boolean flag which is set to false at the beginning of the function, and set to true here?

Reply:
usually we should not. But if we have case of cascading alter table's we could. I am in favor of keeping this

Reply:
> Is it ever possible that `activeAlterTables > 1`?
> 

Yes, if there is a trigger that gets fired on some `ALTER TABLE` command which itself does another `ALTER TABLE`.

> If not, in order to get rid of pg_try/pg_catch, can we simply use a boolean flag which is set to false at the beginning of the function, and set to true here?

I didn't get it. It seems to me even with a boolean variable we need to reset it to `false` after this function is done, so we need to `PG_TRY()`/`PG_CATCH()`?